src/website/challonge.ts

```diff
@@ -380,19 +380,54 @@ export class WebsiteWrapperChallonge implements WebsiteWrapper {
 		return (await this.validateResponse(response)) as ChallongeParticipant[];
 	}
 
-	private wrapPlayer(p: ChallongeParticipant): WebsitePlayer {
+	private async setDiscordId(tournamentId: string, playerId: number, newId: string): Promise<void> {
+		const response = await fetch(`${this.baseUrl}tournaments/${tournamentId}/participants/${playerId}.json`, {
+			method: "PUT",
+			body: JSON.stringify({
+				participant: {
+					misc: newId
+				}
+			}),
+			headers: { "Content-Type": "application/json" }
+		});
+		await this.validateResponse(response);
+	}
+
+	private async getDummyId(
+		player: ChallongeParticipant,
+		tournamentId: string,
+		players?: ChallongeParticipant[]
+	): Promise<string> {
+		if (!players) {
+			players = await this.indexPlayers(tournamentId);
+		}
+		let i = 0;
+		let candidateId = `DUMMY${i}`;
+		while (players.find(p => p.participant.misc === candidateId)) {
+			i++;
+			candidateId = `DUMMY${i}`;
+		}
+		await this.setDiscordId(tournamentId, player.participant.id, candidateId);
+		return candidateId;
+	}
```

Comment:
For a `get` method this is chock-full of side effects...

Reply:
Good point, I ought to rename it